docs/rfc/rfc-1.adoc

```diff
@@ -180,12 +179,48 @@ simple as "these funds can be spent by wallet 0xabc", or in our case, as
 complex as "these funds can be spent by wallet 0xabc but if they aren't spent
 within 30 days they can be spent by wallet 0x123". This gives us the ability to
 create deposits that automatically are refunded after 30 days if they aren't
-swept. Thus, if a user misfunds or they get cold feet (for any reason), all
-they need to do is not submit their reveal and wait 30 days.
+<<sweeping,swept>>. Thus, if a user misfunds or they get cold feet (for any
+reason), all they need to do is not submit their reveal and wait 30 days.
+
+[[sweeping]]
+==== Sweeping
+The operators sign a transaction that unlocks all revealed deposits, combines
+them into a single UTXO with the existing UTXO, and relocks that transactions
+without a 30-day refund clause to same wallet. This has two main effects: it
+consolidates the UTXO set and it disables the refund.
+
+This process is called a "sweep", and occurs based on a governable frequency.
+Any deposit below a governable threshold is ignored.
+
+After submitting the transaction, it will cost some amount of bitcoin based on
+what miners are charging for the bitcoin fee. This fee comes out of the unswept
+deposits. When the transaction clears, the users unswept balances are
+decreased, and their swept balances are increased (after paying their share of
+the mining fee).
```

Comment:
The wallet needs to submit a transaction proving the sweep happened and set/increase users' swept balances. This transaction is expensive. How do we select which operator from the wallet is responsible for submitting it? How do we incentivize it? 

From our earlier discussions, we had an idea of *maintainers* who do not necessarily need to have any stake and their responsibility is just to execute transactions on-chain and get rewards for that. This can even be an additional mode of keep client to make it simple - an optional module one needs to enable/disable with a flag. Then, regular operators do not even need to have any ETH for a day-to-day work.

Still, there is a question of how these maintainers should work. How they are paid and if there is any order in which they are expected to act so that they do not have to race and waste gas on reverted transactions because they compete with each other and because some MEV bot keeps frontrunning everyone.

It could be a good idea to see how Keep3rs works in Yearn and thinking if we can maybe implement a similar ordering mechanism as for relay entry / DKG result submission in random beacon v2, see [here](https://github.com/keep-network/keep-core/blob/main/docs/rfc/rfc-19-random-beacon-v2.adoc#fees-and-rewards). E.g. there is a pool of maintainers, they are getting rewards only for successful transactions, and if the given maintainer missed their chance, they are removed from the pool for a long time. Those who are not in the pool are not eligible to do the maintainer's work. The pool has some initialization period and/or locks some T deposit (or uses stake) to prevent griefing.